tests/repr/invalid-repr-2.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(unexpected_cfgs)]
```

Comment:
Please add a comment on why this is needed.

Reply:
After spending time to understand what this test is trying to do with the missing feature check, I realised that it should just be using `all()` instead, so I’ve changed it to do that instead.